File: Math/prime_sieve.py
# n이하 소수의 리스트 반환
def sieve(n):
    primes = [True] * (n + 1)
    primes[0] = primes[1] = False

    for i in range(2, int(n**0.5) + 1):
        if primes[i]:
            for j in range(i * i, n + 1, i):
                primes[j] = False

    return [x for x in range(2, n + 1) if primes[x]]

File: Math/prime_sieve.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> sieveOfEratosthenes(int n) {
  vector<bool> isPrime(n + 1, true);
  vector<int> primes;

  for (int p = 2; p * p <= n; p++) {
    if (!isPrime[p])
      continue;
    for (int i = p * p; i <= n; i += p)
      isPrime[i] = false;
  }

  for (int p = 2; p <= n; p++) {
    if (isPrime[p])
      primes.push_back(p);
  }

  return primes;
}

int main() {
  vector<int> primes = sieveOfEratosthenes(50);
  for (int i = 0; i < primes.size(); i++)
    cout << primes[i] << "\n";

  return 0;
}